src/Dfe.Spi.GiasAdapter.Infrastructure.GiasSoapApi/GetEstablishmentMessageBuilder.cs

```diff
@@ -0,0 +1,19 @@
+using System.Xml.Linq;
+
+namespace Dfe.Spi.GiasAdapter.Infrastructure.GiasSoapApi
+{
+    internal class GetEstablishmentMessageBuilder : GiasSoapMessageBuilder<GetEstablishmentRequest>
+    {
+        public GetEstablishmentMessageBuilder(string username, string password, int messageValidForSeconds = 30)
+            : base(username, password, messageValidForSeconds)
+        {
+        }
+
+        protected override XElement BuildBody(GetEstablishmentRequest parameters)
+        {
+            return new XElement(soapNs + "Body",
+                new XElement(giasNs + "GetEstablishment",
+                    new XElement(giasNs + "Urn", parameters.Urn)));
+        }
+    }
+}
```

Comment:
The WCF client not great then? Not a change, just a curious observation...

Reply:
Looking online

a) it was harder to add Timestamp to WCF than hand-roll
b) WCF in .NET Core is not especially mature and has a number of issues across the board

So given the relatively simple SOAP requests, seemed easiest to just deal with them manually